pkg/machineoutput/types.go

```diff
@@ -0,0 +1,59 @@
+package machineoutput
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/openshift/odo/pkg/log"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Kind is what kind we should use in the machine readable output
+const Kind = "Error"
+
+// APIVersion is the current API version we are using
+const APIVersion = "odo.openshift.io/v1alpha1"
+
+// Error for machine readable output error messages
+type Error struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Message           string `json:"message"`
+}
+
+// Success same as above, but copy-and-pasted just in case
+// we change the output in the future
+type Success struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Message           string `json:"message"`
+}
+
+// OutputSuccess outputs a "successful" machine-readable output format in json
+func OutputSuccess(machineOutput Success) {
+	printableOutput, err := json.Marshal(machineOutput)
+
+	// If we error out... there's no way to output it (since we disable logging when using -o json)
+	if err != nil {
+		fmt.Fprintf(log.GetStderr(), "Unable to unmarshal JSON: %s", err.Error())
+		os.Exit(1)
+	}
+
+	fmt.Fprintf(log.GetStdout(), string(printableOutput))
+	os.Exit(0)
+}
+
+// OutputError outputs a "successful" machine-readable output format in json
+func OutputError(machineOutput Error) {
+	printableOutput, err := json.Marshal(machineOutput)
+
+	// If we error out... there's no way to output it (since we disable logging when using -o json)
+	if err != nil {
+		fmt.Fprintf(log.GetStderr(), "Unable to unmarshal JSON: %s", err.Error())
+		os.Exit(1)
+	}
+
+	fmt.Fprintf(log.GetStderr(), "%s", string(printableOutput))
+	os.Exit(1)
```

Comment:
Maybe throw the error back instead of exiting here

Reply:
The reason it's implemented this way is in order to manually print out the json output and error out correctly. See line: https://github.com/openshift/odo/pull/1916/files/08e8c6d2ee51eee1b2bf173a2b8ac446d9993466#diff-fd43f0ae9f3ecab43554cd975c2d3761R28

We override the `LogErrorAndExit` function essentially.

What do you think @kadel / @girishramnani ? Keep this here or add it to LogErrorAndExit?